### PR TITLE
Support Python 3.10.

### LINKED ISSUE #270: Support Python 3.10 in CI
Blocked by:

- https://github.com/actions/virtual-environments/issues/4226
- https://github.com/numpy/numpy/issues/20045
- https://github.com/numpy/numpy/issues/20039
- https://github.com/scipy/scipy/issues/14738
- https://github.com/scipy/scipy/issues/14567

Already updated base `python-multi` image with Python 3.10 support: https://github.com/dhermes/python-multi/pull/2

### PR DESCRIPTION
**Pull request by dhermes**
Fixes #270.

### CODE REVIEW

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -27,7 +27,7 @@ jobs:
         key: ubuntu-tests-pip
     - uses: actions/checkout@v2
     - name: Pull Docker image used for CI
-      run: docker pull dhermes/bezier:latest
+      run: docker pull dhermes/bezier:issue-270
```
Comment (dhermes) [inline .github/workflows/linux.yml:30]: TODO: revert this to `dhermes/bezier:latest`

File: .github/workflows/macos.yml [hunk 2]
```diff
@@ -24,11 +24,12 @@ jobs:
       PY_VERSION: '${{ matrix.python-version }}'
       PY_BIN_DIR: '/Library/Frameworks/Python.framework/Versions/${{ matrix.python-version }}/bin'
     strategy:
+      fail-fast: false  # Temporary
```
Comment (dhermes) [inline .github/workflows/macos.yml:27]: TODO: revert this

File: .github/workflows/windows.yml [hunk 3]
```diff
@@ -25,15 +25,16 @@ jobs:
       MINGW_32: 'C:\msys64\mingw32\bin'
       MINGW_64: 'C:\msys64\mingw64\bin'
     strategy:
+      fail-fast: false  # Temporary
```
Comment (dhermes) [inline .github/workflows/windows.yml:28]: TODO: revert this

File: docs/releases/0.10.0.rst [hunk 4]
```diff
@@ -26,8 +26,8 @@ Breaking Changes
 Build
 ~~~~~
 
--  Integrating ``black`` `code formatter`_ into the ``nox -s lint`` session
-   to ensure consistent formatting
+-  Integrating ``black`` `code formatter`_ into the ``nox --session lint``
+   session to ensure consistent formatting
```
Comment (dhermes) [inline docs/releases/0.10.0.rst:30]: TODO: revert this

File: docs/releases/0.10.0.rst [hunk 5]
```diff
@@ -86,8 +86,8 @@ Miscellany
       pre-built Docker container
       (`7634779 <https://github.com/dhermes/bezier/commit/763477958c73a4eb6ce0f89b6b37887c66c10706>`__).
 
--  Added ``nox -s validate_functional_test_cases`` session to ensure that
-   functional test cases always adhere to the JSON schema.
+-  Added ``nox --session validate_functional_test_cases`` session to ensure
+   that functional test cases always adhere to the JSON schema.
```
Comment (dhermes) [inline docs/releases/0.10.0.rst:90]: TODO: revert this

File: noxfile.py [hunk 6]
```diff
@@ -28,38 +28,45 @@
...
+    "pycobertura": "pycobertura >= 2.1.0",
     "Pygments": "Pygments",
     "pylint": "pylint >= 2.11.1",
-    "pytest": "pytest >= 6.2.2",
+    "pytest": "pytest >= 6.2.5",
     "pytest-cov": "pytest-cov",
-    "scipy": "scipy >= 1.6.0",
-    "sympy": "sympy >= 1.7.1",
-    "seaborn": "seaborn >= 0.11.1",
+    "scipy": "scipy >= 1.7.1",
```
Comment (dhermes) [inline noxfile.py:49]: TODO: increase this to 1.7.2 (the first to support 3.10); here and elsewhere (critically in the Dockerfile)

File: noxfile.py [hunk 7]
```diff
@@ -164,6 +171,10 @@ def unit(session):
     unit_deps = BASE_DEPS + (DEPS["sympy"],)
     if interpreter == PYPY:
         local_deps = pypy_setup(unit_deps, session)
+    elif IS_WINDOWS and interpreter == "3.10-32":
+        # The SciPy project hasn't yet shipped a wheel supporting 32-bit
+        # Python 3.10 on Windows.
```
Comment (dhermes) [inline noxfile.py:176]: TODO: file an issue to make sure this is tracked

File: scripts/docker/bezier.Dockerfile [hunk 8]
```diff
@@ -35,12 +38,12 @@ RUN mkdir ${WHEELHOUSE}
 RUN set -ex \
   && virtualenv --python=pypy3 pypy3-env \
   && pypy3-env/bin/python -m pip install --upgrade pip wheel \
-  && pypy3-env/bin/python -m pip wheel --wheel-dir=${WHEELHOUSE} "numpy == 1.20.1" \
+  && pypy3-env/bin/python -m pip wheel --wheel-dir=${WHEELHOUSE} "numpy == 1.21.3" \
   && pypy3-env/bin/python -m pip install ${WHEELHOUSE}/numpy*.whl \
-  && pypy3-env/bin/python -m pip wheel --wheel-dir=${WHEELHOUSE} "scipy == 1.6.0" \
+  && pypy3-env/bin/python -m pip wheel --wheel-dir=${WHEELHOUSE} "scipy == 1.7.1" \
```
Comment (dhermes) [inline scripts/docker/bezier.Dockerfile:43]: 1.7.2

File: setup.py [hunk 9]
```diff
@@ -61,14 +61,11 @@
...
-REQUIREMENTS = ("numpy >= 1.20.1",)
+REQUIREMENTS = ("numpy >= 1.21.3",)
 # See: https://www.python.org/dev/peps/pep-0508/
 #      Dependency specification for Python Software Packages
 EXTRAS_REQUIRE = {
-    "full": ["matplotlib >= 3.3.4", "scipy >= 1.6.0", "sympy >= 1.7.1"],
-    ':python_version<"3.8" and platform_system=="Windows"': [
-        "importlib-metadata"
-    ],
+    "full": ["matplotlib >= 3.4.3", "scipy >= 1.7.1", "sympy >= 1.9"],
```
Comment (dhermes) [inline setup.py:68]: 1.7.2

File: noxfile.py (hunk 7 above)
Comment (dhermes) [reply noxfile.py:176]: https://github.com/dhermes/bezier/issues/274

**Comment by povilasb** [comment]
@dhermes much appreciated 🙌 

Any plans to release it soon? :)

**Comment by dhermes** [comment]
@povilasb Hopefully in the next week or so, thanks!

**Pull request merged**